test/core/test_promise.c

```diff
@@ -509,8 +545,10 @@ int main() {
   em_promise_t test5 =
     emscripten_promise_then(test4, test_all_settled, fail, (void*)5);
   em_promise_t test6 = emscripten_promise_then(test5, test_any, fail, (void*)6);
+  em_promise_t test7 =
+    emscripten_promise_then(test6, test_race, fail, (void*)7);
   em_promise_t assert_stack =
-    emscripten_promise_then(test6, check_stack, fail, NULL);
+    emscripten_promise_then(test7, check_stack, fail, NULL);
```

Comment:
I wonder if this kind of test that has many phases/sub-tests, would make sense written in gtest?   I added one test recently that uses gtest, but perhaps it would be worth looking into using it more?   Once downside is that its maybe a little heavy weight and relies heavily on C++ (so is not so good for testing low level C libraries perhaps?)>

Reply:
gtest is great, but another problem for these tests in particular is that I don't think gtest would play well with the asynchrony happening here. The actual test checks for all of these tests aren't being run until well after `main` returns.